### PR TITLE
Improve modify_user function

### PR DESCRIPTION
**Pull request by ArnoStiefvater**
**What**:

The `comment`,` auth_source` and `group_ids` parameters got added for gmpv7. Now its possible to change those fields too. `auth_source` has to be used explicitly if not the default source allowed for authentication for the user is to be used.
`modify_user` now only takes an ID for identifying the user which is to be modified for gmpv214.
The `name` parameter replaces the old `new_name` parameter for gmpv214 and is not used for identifying a user anymore.

**Why**:

To be able to set the appropriate fields.

**How**:

Added tests.

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

**Comment by y0urself** [comment]
As mentioned before: I think this should be in an earlier version of gmp ...?!

**Comment by ArnoStiefvater** [comment]
> As mentioned before: I think this should be in an earlier version of gmp ...?!

But this change completely breaks the API. For example it is not possible to identifying a user bey name anymore. Its not only adding stuff. I can modify the 208 function to include the comment, user_auth and group_ids params, but the name change breaks the API.

File: gvm/protocols/gmpv7/gmpv7.py [hunk 1]
```diff
@@ -6070,27 +6070,45 @@ def modify_user(
...
+        comment: Optional[str] = None,
         password: Optional[str] = None,
+        auth_source: Optional[UserAuthType] = None,
         role_ids: Optional[List[str]] = None,
         hosts: Optional[List[str]] = None,
         hosts_allow: Optional[bool] = False,
         ifaces: Optional[List[str]] = None,
-        ifaces_allow: Optional[bool] = False
+        ifaces_allow: Optional[bool] = False,
+        group_ids: Optional[List[str]] = None,
```
Comment (y0urself) [inline gvm/protocols/gmpv7/gmpv7.py:6081]:
```suggestion
        group_ids: Optional[List[str]] = None
```

**Review by y0urself — CHANGES_REQUESTED** [review]

**Comment by y0urself** [comment]
you  may add a test for the new Enum `UserAuthType` ? In `testtypes`?

**Review by nichtsfrei — APPROVED** [review]

**Review by y0urself — APPROVED** [review]

**Pull request merged**